src/Providers/Models/DTO/ModelRequirements.php

```diff
@@ -88,6 +91,192 @@ public function getRequiredOptions(): array
         return $this->requiredOptions;
     }
 
+    /**
+     * Checks whether the given model metadata meets these requirements.
+     *
+     * @since 0.2.0
+     *
+     * @param ModelMetadata $metadata The model metadata to check against.
+     * @return bool True if the model meets all requirements, false otherwise.
+     */
+    public function areMetBy(ModelMetadata $metadata): bool
+    {
+        // Check if all required capabilities are supported
+        foreach ($this->requiredCapabilities as $requiredCapability) {
+            $supported = false;
+            foreach ($metadata->getSupportedCapabilities() as $supportedCapability) {
+                if ($supportedCapability->equals($requiredCapability)) {
+                    $supported = true;
+                    break;
+                }
+            }
+            if (!$supported) {
+                return false;
+            }
+        }
+
+        // Check if all required options are supported with the specified values
+        foreach ($this->requiredOptions as $requiredOption) {
+            $optionSupported = false;
+            $supportedOptions = $metadata->getSupportedOptions();
+
+            foreach ($supportedOptions as $supportedOption) {
+                if ($supportedOption->getName()->equals($requiredOption->getName())) {
+                    // Check if the required value is supported by this option
+                    if ($supportedOption->isSupportedValue($requiredOption->getValue())) {
+                        $optionSupported = true;
+                        break;
+                    }
+                }
+            }
+
+            // If no supported options at all, this is only OK if we have no required options
+            if (!$optionSupported) {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    /**
+     * Creates ModelRequirements from prompt data and model configuration.
+     *
+     * @since 0.2.0
```

Comment:
While we will _very likely_ add this to 0.2.0, please always use `n.e.x.t` for any new code. This leaves the version up for a future decision, and we can replace all `n.e.x.t` instances with the correct version number right before a new release.